device/common/include/traccc/seeding/device/count_doublets.hpp

```diff
@@ -28,14 +28,18 @@ namespace traccc::device {
 /// @param[in] config        Seedfinder configuration
 /// @param[in] sp_view       The spacepoint grid to count doublets on
 /// @param[in] sp_ps_view    Prefix sum for iterating over the spacepoint grid
-/// @param[out] doublet_view Container storing the number of doublets
+/// @param[out] doublet_view Collection storing the number of doublets for each
+/// spacepoint
+/// @param[out] nMidBot      Total number of middle-bottom doublets
+/// @param[out] nMidTop      Total number of middle-top doublets
 ///
 TRACCC_HOST_DEVICE
 inline void count_doublets(
-    std::size_t globalIndex, const seedfinder_config& config,
+    const std::size_t globalIndex, const seedfinder_config& config,
```

Comment:
This is purely a personal preference, but when passing something by value, I prefer to:
  - Not put `const` into the declaration into the function;
  - Put `const` into the implementation of the function.

Of course if these two are not separated, then you would put `const` into the one place where the function appears. But in this case I'd only put `const` into the `.ipp` file.

My justification is that making a "by value" argument `const` is an implementation detail that the users of the function are not affected by. Exactly for this reason the compilers allow you to leave off the `const` declaration from the function's API.